extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -278,4 +326,162 @@ public void transform(TransformationContext ctx) {
+            CachedHealthUI cached = liveReload.getContextObject(CachedHealthUI.class);
+            boolean extractionNeeded = cached == null;
+
+            if (cached != null && !cached.cachedHealthPath.equals(healthPath)) {
+                try {
+                    FileUtil.deleteDirectory(Paths.get(cached.cachedDirectory));
+                } catch (IOException e) {
+                    LOG.error("Failed to clean Health UI temp directory on restart", e);
+                }
+                extractionNeeded = true;
+            }
+            if (extractionNeeded) {
+                if (cached == null) {
+                    cached = new CachedHealthUI();
+                    liveReload.setContextObject(CachedHealthUI.class, cached);
+                    Runtime.getRuntime().addShutdownHook(new Thread(cached, "Health UI Shutdown Hook"));
+                }
+                try {
+                    AppArtifact artifact = getHealthUiArtifact(curateOutcomeBuildItem);
+                    Path tempDir = Files.createTempDirectory(TEMP_DIR_PREFIX).toRealPath();
+                    extractHealthUi(artifact, tempDir);
+                    cached.cachedDirectory = tempDir.toAbsolutePath().toString();
+                    cached.cachedHealthPath = healthPath;
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            Handler<RoutingContext> handler = recorder.uiHandler(cached.cachedDirectory,
+                    httpRootPath.adjustPath(health.rootPathUi));
+            routeProducer.produce(new RouteBuildItem(health.rootPathUi, handler));
+            routeProducer.produce(new RouteBuildItem(health.rootPathUi + "/*", handler));
+            notFoundPageDisplayableEndpointProducer
+                    .produce(new NotFoundPageDisplayableEndpointBuildItem(health.rootPathUi + "/"));
+        } else if (health.alwaysIncludeUi) {
+            AppArtifact artifact = getHealthUiArtifact(curateOutcomeBuildItem);
+            //we are including in a production artifact
+            //just stick the files in the generated output
+            //we could do this for dev mode as well but then we need to extract them every time
+            for (Path p : artifact.getPaths()) {
+                File artifactFile = p.toFile();
+                try (JarFile jarFile = new JarFile(artifactFile)) {
+                    Enumeration<JarEntry> entries = jarFile.entries();
+
+                    while (entries.hasMoreElements()) {
+                        JarEntry entry = entries.nextElement();
+                        if (entry.getName().startsWith(HEALTH_UI_WEBJAR_PREFIX) && !entry.isDirectory()) {
+                            try (InputStream inputStream = jarFile.getInputStream(entry)) {
+                                String filename = entry.getName().replace(HEALTH_UI_WEBJAR_PREFIX + "/", "");
+                                byte[] content = FileUtil.readFileContents(inputStream);
```

Comment:
I would suggest incorporating this line in the if statement on the following line so you can avoid some processing which will be discarded for ignored files.

Reply:
It's used later (outside of the if) - see line 416